packages/next/src/experimental/testmode/playwright/index.ts

```diff
@@ -22,7 +25,7 @@ export function defineConfig<T extends NextOptionsConfig, W>(
 export function defineConfig<T extends NextOptionsConfig = NextOptionsConfig>(
   config: base.PlaywrightTestConfig<T>
 ): base.PlaywrightTestConfig<T> {
-  return base.defineConfig<T>(config)
+  return base.defineConfig<T>(Object.assign(defaultPlaywrightConfig, config))
```

Comment:
1. You can't assign into `defaultPlaywrightConfig`. At a minimum, it should be copied. Which can be done simply as `{...defaultPlaywrightConfig}`. Otherwise you'd be mutating a constant. To avoid this kind of mistake, you could also call `Object.freeze()` on that constant.
2. `Object.assign` is not enough, IMHO. E.g. we can't just override `use: {baseURL, ...}` completely - these are completely unrelated values to each other. At least the case could be made that something like `webServer: {}` is atomic and goes together, but not for `use`.

Instead of using some generic `merge`, I'd simply recommend building merge here with the knowledge of what objects mean here. Here's how you can do this:


```
function defineConfig<...>(config: ...): {
  return base.defineConfig<T>({
    ...defaultPlaywrightConfig,
    ...config,
    use: {
      ...defaultPlaywrightConfig.use,
      ...config.use,
    },
    // And so on.
  });
}
```

This is slightly more verbose, but otherwise very clear and self-explanatory.

Reply:
done